src/coreComponents/mesh/utilities/ComputationalGeometry.hpp

```diff
@@ -509,6 +509,54 @@ real64 pyramidVolume( real64 const (&X)[5][3] )
   return elementVolume< finiteElement::H1_Pyramid_Lagrange1_Gauss5 >( X );
 }
 
+/**
+ * @brief Compute the volume of a prism with N-sided polygon base
+ * @tparam N the number of sides in the polygon base
+ * @param[in] X vertices of the prism
+ * @return the volume of the prism
+ *
+ * @note The volume is computed splitting the prism into wedges
+ */
+template< integer N >
```

Comment:
Yes, good suggestion. I would restrict the use of this function only to `N > 4`, i.e. for pentagonal prisms and above, and rely on the other methods for wedges and hexahedra. The same `static_assert` must be added here

https://github.com/GEOSX/GEOSX/blob/2a5c45b1193d87259710c278407d60246c766fbb/src/coreComponents/mesh/generators/PrismUtilities.hpp#L30-L33

to prevent use for wedges and hexahedra (local indices for faces would be wrong).